### PR TITLE
refactor(git) removed 'Impl' suffix as it's is just noise keep it DRY

### PR DESCRIPTION
**Pull request by edewit**

### CODE REVIEW

File: services/git-service-impl/src/main/java/io/fabric8/launcher/service/gitlab/GitLabService.java [hunk 1]
```diff
@@ -42,7 +41,7 @@
 /**
  * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
  */
-class GitLabServiceImpl extends AbstractGitService implements GitLabService {
+class GitLabService extends AbstractGitService implements GitService {
```
Comment (gastaldi) [inline services/git-service-impl/src/main/java/io/fabric8/launcher/service/gitlab/GitLabService.java]: GitLabGitService?

File: web/src/main/java/io/fabric8/launcher/web/producers/GitServiceProducer.java [hunk 2]
```diff
@@ -31,7 +30,7 @@
     private Instance<GitServiceFactory> gitServiceFactories;
 
     @Inject
-    private GitHubServiceFactory defaultProvider;
+    private GitServiceFactory defaultProvider;
```
Comment (gastaldi) [inline web/src/main/java/io/fabric8/launcher/web/producers/GitServiceProducer.java]: This MUST inject the GitHub factory. It is the fallback service

**Comment by edewit** [comment]
@gastaldi updated

File: core/core-impl/src/test/java/io/fabric8/launcher/core/impl/MissionControlIT.java [hunk 3]
```diff
@@ -64,7 +64,7 @@
     private OpenShiftServiceFactory openShiftServiceFactory;
 
     @Inject
-    private GitHubServiceFactory gitHubServiceFactory;
+    private GitServiceFactory gitServiceFactory;
```
Comment (gastaldi) [inline core/core-impl/src/test/java/io/fabric8/launcher/core/impl/MissionControlIT.java]: I'm curious: Does this test pass? AFAIK doesn't have any produces for GitServiceFactory

**Comment by gastaldi** [comment]
We'd need to have new qualifiers (like @GitHub, @Bitbucket and @Gitlab) for each GitServiceFactory implementation if we're not having different types anymore

**Comment by gastaldi** [comment]
[test]

**Comment by gastaldi** [comment]
Or an enum with the supported types and a `@GitProvider(GITHUB)` qualifier, maybe that would be better. The /services/git/providers would just need to return the enum values

**Comment by gastaldi** [comment]
Replaced by https://github.com/fabric8-launcher/launcher-backend/pull/154

**Comment by gastaldi** [comment]
@edewit I moved your code to an `edewit-impl-remove` branch inside the launcher-backend repository, so we can both work on it. IT are not yet passing

**Comment by gastaldi** [comment]
Fixed the tests. Will merge that other PR once the tests pass. Thanks!

**Pull request closed without merging**